### ISSUE TITLE
Collection item labels

### ISSUE THREAD

**Issue by SaschaAdler** [crkn-rcdr/Access-Platform#40]
I think we've forgotten about something in the current platform that we're going to need to support.

Series issues currently use a combination of what's known in copresentation as `label` and `plabel` to generate their public display labels. `plabel` is the label of the issue's series. This allows the issue's public display label to comprise both the name of the series and the name of the issue [when viewing it directly](https://www.canadiana.ca/view/oocihm.N_00367_19100513/1?r=0&s=1), but only comprise the issue label [when viewing the series](https://www.canadiana.ca/view/oocihm.N_00367).

**I think it is vitally important that the inherent label of any manifest provide its entire context.** As such, the manifest for an issue should contain both the `plabel` and the `label` of the current issue record. Instead of injecting the navigation to the issue's parent collection inside the label, as is currently done on most of our portals, [the link can be displayed adjacent to the display label, as is done on the Parliament portal](https://parl.canadiana.ca/view/oop.bills_SOCHOC_1205_1/1?r=0&s=1).

It would still be useful to allow for the shortening of issue labels to just `label` in the context of viewing the series collection. I propose a change to the collection database as follows:

* the `items` array becomes an array of objects, with key `id` corresponding to the item's noid
* an optional key, `itemLabel`, can be specified, which can contain the shortened label for the item in the collection context
* the item's actual label is still looked up, and if `itemLabel` does not exist it is displayed

As far as current issue records go, we will need to update their labels to add the series record's label. I believe Smelter already does some internalmeta lookups for now to make this possible, but I'm not sure of any alternatives.

**Comment by RussellMcOrmond** [comment]
In the current platform it is Press that brings those labels together in the same record for the search/presentation documents, and not something that is maintained in the source (internalmeta).  I don't know that we want to try to do away with the concept of "Press", and treat the manifest/collection databases as presentation/search documents.  Implementing the functionality of Press within the API seems to be adding unnecessary complexity to both the API and the manifest/collection databases.



We have a policy question to determine before we can decide how to implement this: Is a manifest is allowed to be a member of more than one ordered collection?

With issues of a series it had a parent, and it was a string field that didn't support the concept of there being more than one parent.  Press would look up the parent to pull in label information for search and presentation documents.

We could decide that a manifest document can have a parent noid field that is managed by the collection manager, but we would still need to have a way to indicate to the collection manager that it is a "special" ordered collection it is editing that should maintain the concept of a single parent (and provide warnings/errors if a manifest that already has a parent is attempted to be added).

If a manifest can be a member of more than one ordered collection, then Press can offer an array of references to parents (calculated at Press time), with CAP needing to be updated to handle that new concept.  This was in fact what I was assuming we were planning on doing, allowing staff to manage those relationships as they see fit rather than trying to implement the old system within the new.  If they decided to only ever have a manifest in a single ordered collection, then the array in the presentation document would only ever have one element.


* Smelter does/should not look at any of the old-platform databases like 'internalmeta', 'copresentation', or 'cosearch'.   It only interacts with 'dipstaging', 'manifest' and AIP's in Swift (including via Cantaloupe to get image dimensions/etc).   Smelter does/should not involve itself with the concept of collections (ordered or otherwise).  That is part of the point of P-A split, which is to not have those access relationships thought of as having anything to do with preservation.

**Comment by RussellMcOrmond** [comment]
In my reply I broke my own "prime directive" which is to stick to the specific project, and not get paralysed by trying to think too many steps ahead.  We should ensure our design is extensible, but we should not be concerned with implementing future features in the current design or software.

[Saga: Preservation/Access Split (CRKN staff access only)](https://crkn.sharepoint.com/:w:/s/platform/EYgmO_v7OYJHnzQj8cgurw8Bjn-GS2vaUxpMsM2bxpxXlA?e=hiLgx9)

This has a list of "In Scope" and "Out of Scope" and we should be trying to keep to those.  If we require we diverge then we should discuss and change the documentation to reflect.

So, to this specific issue:

* A manifest can be in one and only one ordered collection, and if it is in an ordered collection it cannot simultaneously be in any other collection.  This mirrors in the new databases the structure that existed in the old.
* Currently Press will generate an error if the parent of an issue record cannot be found. See the [List of Press failures](https://admin.canadiana.ca/couchview/internalmeta/tdr/presss?endkey=%5B0%5D&inclusive=true&keytype=json&range=true&reduce=false&startkey=%5B0%5D), which are currently all failures to find the parent.
* The new Press would generate an error if a manifest was in an ordered collection and any other collection (the manifest will not be available online).
* This is separate from the work Press would do to build the array for what collections its parent collection is in.  The parent could be in many collections, even if the child manifest can only be in a single ordered collection.


I believe having the collection editor and/or the API contain the logic that is currently within Press (relating to manifest parentage, etc) constitutes a new feature, and should be declared out of scope for the P-A separation saga.  We can discuss and potentially schedule that new feature at a later date, once P-A separation is launched and the old parallel databases are gone.

**Comment by SaschaAdler** [comment]
We currently have manifests in multiple ordered collections; namely, Parliament manifests. These ordered collections have yet to be imported into the collection database because they aren't series records, but they're a currently supported logical concept in our platform. Since "Changes to CAP so that it uses the Access Repository for indexing, description, and content delivery" is in scope, we will need to be able to use these concepts as collections before we've completed the split (i.e., replaced the data in internalmeta with something else).

I agree that Smelter shouldn't look at internalmeta, but is my understanding correct that loadseries does? It can also perform what I consider to be the necessary step of prepending `plabel` to `label`, and updating the series collection with the `itemLabel`s as I've described above. This can happen when loadseries fills the collection's item list. In my opinion, this is the only way we can finally rid ourselves of "series record" as a concept, something which is certainly a goal if it is technically not required for the split MVP.
